### PR TITLE
Remove cocotb_bus dependency from regression

### PR DESCRIPTION
**Pull request by ktbarrett**
Remove use of cocotb_bus components from the regression tests.
Moves many tests and examples to the cocotb_bus repo (https://github.com/cocotb/cocotb-bus/pull/20).
Requires refactoring the matrix_multiplier to not use the matrix_multiplier. I took the opportunity to update it to use queues.
Also removes the cocotb_bus import stubs.

### CODE REVIEW

File: examples/Makefile [hunk 1]
```diff
@@ -28,10 +28,7 @@
 ###############################################################################
 
 EXAMPLES := adder/tests \
-            axi_lite_slave/tests \
-            dff/tests \
             simple_dff \
-            endian_swapper/tests \
             matrix_multiplier/tests \
             mean/tests \
```
Comment (marlonjames) [inline examples/Makefile:33]:
```suggestion
```

**Comment by marlonjames** [comment]
> Requires refactoring the matrix_multiplier to not use the matrix_multiplier.

😆

**Review by marlonjames — COMMENTED** [review]
`test_iteration_verilog` relies on endian_swapper file: https://github.com/cocotb/cocotb/pull/2445/checks?check_run_id=2044362799#step:25:3782

`matrix_multiplier` looks good.

**Comment by ktbarrett** [comment]
You mean what I know.

**Comment by marlonjames** [comment]
Re-running CI to see if the coverage works this time.

**Review by marlonjames — APPROVED** [review]

**Pull request merged**